### DISCUSSION TITLE
Pushing local repo to GitHub

### DISCUSSION THREAD

**Discussion by AztecWarlord** [Cyfrin/foundry-full-course-cu#84] [Q&A] [answered]
I am trying to push my local repo to GitHub repo but keep getting this error:
```
error: src refspec main does not match any
error: failed to push some refs to 'https://github.com/AztecWarlord/foundry-fund-me-f23.git'
```
the issue now is when I try to push I get this error:
```
git push -u origin main
Username for 'https://github.com': AztecWarlord
Password for 'https://AztecWarlord@github.com':
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/AztecWarlord/foundry-fund-me-f23.git/'
```
Any help will be great!

**Comment by jyotirmoydotdev** [comment]
You can use the `gh` package to clone your repository and push code changes to GitHub.

First, install `gh` using the following commands:

```
apt install -y gh
git config --global user.email "you@example.com"
git config --global user.name "Your Name"
```

Then, authenticate your session using `gh auth login`.

Next, clone your repository using the following command:

```
gh repo clone https://github.com/AztecWarlord/foundry-fund-me-f23
```

Move all the code into the `foundry-fund-me-f23` directory and commit the changes with the following commands:

```
cd foundry-fund-me-f23
git add .
git commit -m "message"
git push
```

**Reply by AztecWarlord** [reply]
I am getting this error:
```
E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
```

**Reply by AztecWarlord** [reply]
I got passed the: 
```
apt install -y gh
git config --global user.email "you@example.com"
git config --global user.name "Your Name"
```
but once I got to this part:
```
cd foundry-fund-me-f23
git add .
git commit -m "message"
git push
```
I return to the same Authentication failed error.

**Comment by alymurtazamemon** [comment, marked as answer]
@AztecWarlord You need to authenticate yourself into the terminal. For it, go to the GitHub profile settings, in the left panel at the bottom you will find the developers settings click on it, and then Personal access tokens => Token (classic) => Generate new token (give permission according to your need, you can change them later as well) and now use it as a password in the terminal.

**Reply by AztecWarlord** [reply]
thank you it worked! not sure why I was getting recommendations on google to use a SSH key that gave me a lot of problems but you way was seamless! thanks again!
